aptos-move/framework/aptos-stdlib/sources/cryptography/ed25519.move

```diff
@@ -166,6 +170,43 @@ module aptos_std::ed25519 {
         std::hash::sha3_256(pk_bytes)
     }
 
+    #[test_only]
+    fun generate_keys(): (SecretKey, UnvalidatedPublicKey) {
```

Comment:
Then in the test we have to do an awkward `ValidatedPublicKey`-to-`UnvalidatedPublicKey` conversion, because verification method only accept `UnvalidatedPublicKey`, which also makes sense.

IMO `ValidatedPublicKey` is unnecessary and `UnvalidatedPublicKey` can just be `PublicKey`?

Reply:
That's fine: we'll pay the price and that awkward conversion. It's worth it for type-safety: it's important we distinguish between validated and unvalidated PKs.

Indeed, our current (so-called "strict") signature verification works only for unvalidated PKs / we do not have any APIs for verifying against validated PKs. That's fine also; we might add them later.